src/server/table.h

```diff
@@ -70,6 +76,7 @@ struct DbTable : boost::intrusive_ref_counter<DbTable, boost::thread_unsafe_coun
   absl::flat_hash_map<std::string, std::vector<ConnectionState::ExecInfo*>> watched_keys;
 
   mutable DbTableStats stats;
+  std::array<SlotStats, ClusterConfig::kMaxSlotNum + 1> slots_stats;
```

Comment:
Consider using a `vector` here instead of an `array`. Reasons:
1. In none cluster mode, it would not take up 16k * `sizeofSlotStats` unneeded memory
2. It turns out that, at least sometimes, having an `array` of this size significantly slows down compilation time. Normally I would not care a bit about this, but when I experimented with it in my changes to `ClusterConfig` it took a whopping 5m(!!) to compile just that single file
3. It would allow us to be dynamic in the size we allocated, like for example if we only care about slots `[0, 1'000)`, we don't have to allocate the entire thing. I will say that this is the weakest argument of the 3, and that it definitely won't be a consideration if the above 2 did not exist.

Reply:
I meant that perhaps, in the future, we could only use the ones we care about, but that's not something we should take into consideration now.